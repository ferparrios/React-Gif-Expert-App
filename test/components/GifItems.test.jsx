import { render, screen } from "@testing-library/react"
import { GifItems } from "../../src/components/GifItems"

describe('Pruebas en el <GifItems />', () => {

    const title = 'Saitama'
    const url = 'https://saitama.com/saitama.jpg'

    test('debe de hacer match con el snapshot', () => {
        const {container} = render(
            <GifItems title={title} url={url} />
        )
        expect(container).toMatchSnapshot()
    })

    test('Mostrar la imagen con el url y el alt indicado', () => { 
        render(<GifItems title={title}  url={url} />)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
     })

     test('debe de mostra el titulo en el componente', () => { 
        render(<GifItems title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
      })
})