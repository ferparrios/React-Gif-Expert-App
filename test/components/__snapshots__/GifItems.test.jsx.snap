// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifItems /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
